pilot/pkg/networking/core/configgen.go

```diff
@@ -35,7 +35,7 @@ type ConfigGenerator interface {
 	BuildListeners(node *model.Proxy, push *model.PushContext) []*listener.Listener
 
 	// BuildClusters returns the list of clusters for the given proxy. This is the CDS output
-	BuildClusters(node *model.Proxy, push *model.PushContext) []*cluster.Cluster
+	BuildClusters(node *model.Proxy, push *model.PushContext) ([]*discovery.Resource, model.XdsLogDetails)
```

Comment:
why do we need to return `model.XdsLogDetails`? I wonder if theres a better, cleaner way to do this

Reply:
To bubble up the cache hit/miss details so we can include it in the logs.